### PR TITLE
Add SetCrystalLocation

### LINKED ISSUE #24329: Create SetCrystalLocation algorithm
This will create an algorithm `SetCrystalLocation` that will provide a straightforward interface to change the sample location for event workspaces.

### PR DESCRIPTION
**Pull request by sullivanbt**
New algorithm `SetCrystalLocation` to set the sample position of an events workspace.  This provides a nice, simple interface to adjust sample position when recreating MD workspaces with samples not at the origin.

No tests have been added since it changes the sample position the same way that `OptimizeCrystalPlacement` does, which already has tests.  

**To test:**
```python
events = Load('TOPAZ_3007_bank_37_20_sec.nxs')
sample = mtd['events'].getInstrument().getSample()
print('Sample position before SetCrystalLocation: {}'.format(sample.getPos()))
SetCrystalLocation(InputWorkspace=events, OutputWorkspace=events, NewX=0.1, NewY=0.1, NewZ=0.1)
print('Sample position after SetCrystalLocation: {}'.format(sample.getPos()))
```
You can also verify that MDWorkspaces created from these are different.  Note that if you test that, you must delete the preprocessed detectors workspace or name a new preprocessing workspace when calling `ConvertToMD`.

Fixes #24329 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by VickieLynch — DISMISSED** [review]
Not sure why the doc tests are failing since the file is found by doc tests on my machine.

**Comment by VickieLynch** [comment]
Ubuntu build servers do not have systemtest data.  Change doc test to read this file:  events = Load('BSS_11841_event.nxs')

**Review by VickieLynch — DISMISSED** [review]

File: Framework/Crystal/inc/MantidCrystal/SetCrystalLocation.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2012 ISIS Rutherford Appleton Laboratory UKRI,
```
Comment (soininen) [inline Framework/Crystal/inc/MantidCrystal/SetCrystalLocation.h]: The year should probably be 2018 already.

File: Framework/Crystal/inc/MantidCrystal/SetCrystalLocation.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  const std::string name() const override { return "SetCrystalLocation"; };
+  /// Summary of algorithms purpose
+  const std::string summary() const override {
+    return "This algorithm sets the sample location of the "
+           "input event workspace.";
+  }
+
+  int version() const override { return 1; };
+
+  const std::string category() const override {
```
Comment (soininen) [inline Framework/Crystal/inc/MantidCrystal/SetCrystalLocation.h]: Would it make sense to have a `seeAlso()` method? `OptimizeCrystalPlacement` for example was mentioned in the PR's description.

File: Framework/Crystal/src/SetCrystalLocation.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+// simple algorithm that changes the sample position of the input
+// event workspace.
+void SetCrystalLocation::exec() {
+  EventWorkspace_sptr events = getProperty("InputWorkspace");
+  EventWorkspace_sptr outEvents = getProperty("OutputWorkspace");
+  double newX = getProperty("NewX");
+  double newY = getProperty("NewY");
+  double newZ = getProperty("NewZ");
+  V3D newSamplePos = V3D(newX, newY, newZ);
```
Comment (soininen) [inline Framework/Crystal/src/SetCrystalLocation.cpp]: The variables above could be declared `const`, I believe.

**Review by soininen — CHANGES_REQUESTED** [review]
A few minor comments below. Why no unit tests? Release notes should be added, as well.

**Comment by sullivanbt** [comment]
All good suggestions - the newest version takes them all into account.

**Review by soininen — APPROVED** [review]

**Comment by soininen** [comment]
Looks good to me. The failing Windows build looks unrelated.

**Pull request merged**